src/backend/distributed/commands/statistics.c

```diff
@@ -248,3 +302,34 @@ GetExplicitStatisticsIdList(Oid relationId)
 
 	return statisticsIdList;
 }
+
+
+/*
+ * GetRelIdByStatName returns the relation id for given statistics name.
+ * If statistics doesn't exist, returns InvalidOid.
+ */
+static Oid
+GetRelIdByStatName(List *statNameList)
+{
+	PG_TRY();
+	{
+		Oid statsOid = get_statistics_object_oid(statNameList, false);
+		HeapTuple tup = SearchSysCache1(STATEXTOID, ObjectIdGetDatum(statsOid));
+
+		if (!HeapTupleIsValid(tup))
+		{
+			/* should not happen */
+			elog(ERROR, "cache lookup failed for statistics object %u", statsOid);
+		}
+
+		Form_pg_statistic_ext statisticsForm = (Form_pg_statistic_ext) GETSTRUCT(tup);
+		ReleaseSysCache(tup);
+
+		return statisticsForm->stxrelid;
+	}
+	PG_CATCH();
+	{
+		return InvalidOid;
```

Comment:
PG_CATCH is a sketchy construct. We should only ever use it do some cleanup + PG_RE_THROW(). We've made some exceptions to this rule, but those do have some bugs.

I think we don't need it at all here. We could instead return InvalidOid in the `if (!HeapTupleIsValid(tup))` block.

Reply:
Yes, I already removed that. Will push in a few minutes.